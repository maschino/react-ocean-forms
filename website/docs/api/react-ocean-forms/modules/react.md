---
id: "react"
title: "react"
sidebar_label: "react"
---

Copyright (c) 2018-present, Umweltbundesamt GmbH

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.

## Index

### Type aliases

* [PropsOf](react.md#propsof)

## Type aliases

###  PropsOf

Æ¬ **PropsOf**: *T extends function ? P : T extends object ? P : never*

Acquire the props for a Component {T}
