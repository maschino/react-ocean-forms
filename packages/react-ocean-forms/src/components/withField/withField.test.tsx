import React from 'react';

import { render } from '@testing-library/react';

import { withField } from './withField';
import { Form } from '../Form';

describe('withField', () => {
  it('should render without error', () => {
    const TestComponent: React.FC = () => <div id="test-component" />;
    const WrappedComponent = withField(TestComponent);

    const { asFragment } = render(
      <Form>
        <WrappedComponent name="mockComponent" label="mockComponent" />
      </Form>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
