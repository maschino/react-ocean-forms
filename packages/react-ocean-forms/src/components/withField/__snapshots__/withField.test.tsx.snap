// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withField should render without error 1`] = `
<DocumentFragment>
  <form
    class=""
  >
    <div
      id="test-component"
    />
  </form>
</DocumentFragment>
`;
